Space Ninja V2

Warpzoned opened the pull request:
Basically Space Ninja remastered, a lot of its existing content remains unchanged, this is more of an array of new additions and/or changes that are generally agreed by the community to make for a more enticing (And unique) Space Ninja, includes:

- New core mechanics
- New flavour items
- New abilities
- Expanded objectives
- Balance changes

### Why

Well, for starters, engaging ghost roles are currently missing in the quantity department, and so it's my plan to improve the ones we already have and love.

Space Ninjas at the moment also have some outright broken abilities, as well as extremely linear gameplay due to lacking what they should be best at: Stealthing, which is what this document also addresses.

Warpzoned committed:
initial commit

Warpzoned committed:
written doc done

might change some stuff, won't PR just yet

Warpzoned committed:
formating update and new interactions

Warpzoned committed:
final touches

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+
+All Space Ninjas will now have a Spider Clan Code they will have to live by, and this thus affects gameplay substantially by imposing certain restrictions.
+- The already existing one, being the inability to shoot ranged weapons, will be included in this Code.
+- The second restriction is making it so the Space Ninjas can no longer tend to their own wounds - This includes applying topicals, injecting syringes, consuming pills, and drinking liquid chemicals, as it would be preferred that they are not even spotted during their mission. To compensate, Space Ninjas will now have increased passive regeneration as well as faster bloodloss recovery that far ex ...
```
Will this ability be based on their suit power, turning the ninja into a "live and die on your suit and engineering" or will this be something seen on the actual ninja character?

Another consideration: can a ninja go into crit in a concealed spot and be slowly mended and revived given enough time? The idea is that if a ninja knows they might go crit to bloodloss/wounds soon they would find a spot to go into crit and hopefully come back out.

Warpzoned replied:
> Will this ability be based on their suit power, turning the ninja into a "live and die on your suit and engineering" or will this be something seen on the actual ninja character?

This will be something on the character itself, the suit can only do so much.

> Another consideration: can a ninja go into crit in a concealed spot and be slowly mended and revived given enough time? The idea is that if a ninja knows they might go crit to bloodloss/wounds soon they would find a spot to go into crit and hopefully come back out.

I hadn't considered this, mostly due to the fact that from my point of view it'd be pretty much impossible for the ninja to go into crit from DoTs, although, if they do, yeah I can see them getting stabilizied again after a while.

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+
+## New Items & Mechanics
+
+### Flavour Additions
```
Another possible suggestion is a special Ninja PDA instead of just having an ID card in their pocket willy nilly.

Warpzoned replied:
> Another possible suggestion is a special Ninja PDA instead of just having an ID card in their pocket willy nilly.

agreed! I was also thinking of making the station pinpointer built into it, it's just that it'd probably be kind of a much bigger hurdle to implement than other more aesthethic-focused apparel like the sheath or satchel, hence why I didn't include it.

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+- Equipping the space ninja helmet is now an instantaneous toggle action that comes with the space ninja suit.
+
+### Ability Additions
+- The Space Ninja can now fabricate ninja caltrops, a variant of ninja throwing stars with an emphasis on piercing damage and bleeding, once deployed, that is, dropped on the ground, they will progressively get more and more camouflaged to the point of near-invisibility. Upon being stepped on, the caltrops will stick to the victim and perform continous bloodloss damage until taken off.
```
Extremely cool, I like the idea. It would also be cool if it told the Ninja that the caltrops had been activated (stepped on)

Warpzoned replied:
> Extremely cool, I like the idea. It would also be cool if it told the Ninja that the caltrops had been activated (stepped on)

I discussed about making it so that every person that stepped on the caltrops would have a miniscule tracking implant injected (that'd show on the ninja's own handheld), but that's probably a bit far out of reach, oh and thank you!

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+- A strongbox for Cargo containing all their deposited spesos, can be the target of one of the Space Ninja's objectives, located in the vault - An alternative to "Steal X technologies".
+- An AI law board that can be used for one of the Space Ninja's objectives - An alternative to "Detonate your spider charge".
+- The criminal records computer now only announces the error locally in the Security channel, instead of it being broadcasted as a Central Command announcement station-wide.
+- The communications computer objective now revolves around the Space Ninja sending a Spider Clan Announcement from it, the contents of said announcement being up to their discretion. Life before, it requires the use of turned on space ninja gloves, the one key difference being that now the EMAG interaction is instantaneous, overriding all access restrictions as well as unlocks Spider Clan as an  ...
```
Does this completely remove calling in a threat? I would still like to see calling in threats. Part of the whole sabotage thing.

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+
+### Nerfs
+- [Fix ninja stunlocking](https://github.com/space-wizards/space-station-14/pull/33244)
+- All Space Ninjas are now bio-engineered humans, making it so they don't get specie-specific attributes such as tail-dragging.
```
I'd like to still see player characters, I'm partially against having everyone be human for the sake of balance, even though it's easiest. I know I love being a spider as a ninja just because I'm part of the "Spider Clan" and I RP it as well.

Warpzoned replied:
> I'd like to still see player characters, I'm partially against having everyone be human for the sake of balance, even though it's easiest. I know I love being a spider as a ninja just because I'm part of the "Spider Clan" and I RP it as well.

I disagree, while part of it is for the sake of balance, I find it to also be much more immersive for certain factions to prefer certain species, just like how S.E.L.F. would prefer IPCs, the Spider Clan prefers bio-engineered humans, and honestly, the ninja fit just goes much harder when there's no added characteristics like wings or tails.

ArtisticRoomba replied:
B-b-b-but... spider... spider clan...

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+- Space Ninjas now come outfitted with a ninja headset as per [Alwayswannahunt's PR](https://github.com/space-wizards/space-station-14/pull/32841) , green encryption key included.
+- To compensate for the newly removed EMAG interactions with lathes and printers, the space ninja gloves can now unlock access-restricted lockers and crates.
+- The katana dash is considerably quieter, at the cost of a longer recharge duration.
+- The energy katana's slash rate is slightly faster than it was previously when one-handed to compensate for it's damage loss.
```
So the DPS is unchanged, or is it still less?

ArtisticRoomba reviewed (commented):
Overall a really cool proposal, I think this will definitely push players further into the mindset of a sneaky saboteur instead of just a raw murderboning role like the Lone Operator.

deltanedas commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+### Nerfs
+- [Fix ninja stunlocking](https://github.com/space-wizards/space-station-14/pull/33244)
+- All Space Ninjas are now bio-engineered humans, making it so they don't get specie-specific attributes such as tail-dragging.
+- The space ninja gloves have lost their EMAG interactions with lathes and printers.
```
this was never a thing

deltanedas replied:
its whitelisted to airlocks so ninja was just trolling you

Warpzoned commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+- The energy katana's slash rate is slightly faster than it was previously when one-handed to compensate for it's damage loss.
+- Grants the ability to wield the energy katana for a hard-hitting swing, at the cost of a slower slash rate.
+- The ninja throwing stars and ninja caltrops can be stacked, as well as thrown in spreads.
+- When phased, the Space Ninja is able to discreetly steal from people, at the cost of a longer action duration.
```
> I'd like to see the ninja's phase cloak be significantly buffed to near-invisibility when cloaked given the ninja is moving _stupidly slow_ in order to better enable this interaction.

> I feel like giving ninjas their cloak similar to how stealth boxes behave could be something interesting to look at, but have it be tuned to reflect their role better. Speedy ninja = not as cloaked, slow and steady ninja = very cloaked, extremely hard to see.

same here! I think it'd be super cool if the ninja had a system similar to that of loneops' cardboard boxes, just a tad afraid of how meta it could become for the ninja to just walk _super_ slowly everywhere

ArtisticRoomba replied:
Ninja lives and dies off of speed. Part of the ninja's survivability is the ability to get in and out of basically any situation lightning quick before someone can even _radio_ security, whether they are cloaked or not.

The proposed change can put players in a different mindset:

Do I slowly sneak up to captain and copy their ID, or steal their bag, concealing myself to make my presence unaware to security, or do I go in loud and make a mess, allowing security to know that I exist and cause them to gear up?

There's a lot of different ways you could play this.

Warpzoned replied:
> Do I slowly sneak up to captain and copy their ID, or steal their bag, concealing myself to make my presence unaware to security, or do I go in loud and make a mess, allowing security to know that I exist and cause them to gear up?

this sold me on it, sounds super fun to be able to change your strategy depending on the go with something simple like this too, I'll put it in in a bit👍

Warpzoned commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,61 @@
+- [Fix ninja stunlocking](https://github.com/space-wizards/space-station-14/pull/33244)
+- All Space Ninjas are now bio-engineered humans, making it so they don't get specie-specific attributes such as tail-dragging.
+- The space ninja gloves have lost their EMAG interactions with lathes and printers.
+- Removes the survival box from the ninja's satchel, as there's no longer any use for it.
```
> Should the ninja get advanced tools? They are a hyper advanced, bio-engineered clan after all. It seems silly they have their crazy powerful suit but you're still spending 13 minutes doorhacking.

I think they should yes, the problem is that the direct upgrade to some tools (like crowbars) are immediately the highest grade, unlike the welding tool that has 4 different tiers (normal, industrial, advanced, experimental), so until that is taken care of, I think the current tools are fine (plus, a big thing about ninjas is them climbing up the ladder in terms of cells, keys, and tools)

Warpzoned commented:
thank you!

Warpzoned committed:
formatting update & more

added progressive cloaking
removed the "can no longer emag lathes/printers" nerf because they couldn't in the first place
fixed spelling and capitalizations

Warpzoned committed:
spelling and ability fixes

Warpzoned committed:
objectives expanded 

rewrote objectives

Warpzoned committed:
final update (methinks)

cleared up some additions, added the pda and increased reflect chance when wielding

SpaceRox1244 commented:
I want to repeat this as a comment because I dont know if it is visible as a pending review, but on the mention of all ninjas being humans: 

I dont really see the point in making all of these species who are interesting in their own way and then not using them for antags, an entire class of role in which their unique attributes can shine. It also sounds kind of arbitrary to justify the decision by saying the outfit looks better, or to claim that the faction just prefers humans. Like, im not entirely against any instance of themed faction, but like, Im not sure why they just would prefer humans here.

Warpzoned commented:
> I want to repeat this as a comment because I dont know if it is visible as a pending review, but on the mention of all ninjas being humans:
> 
> I dont really see the point in making all of these species who are interesting in their own way and then not using them for antags, an entire class of role in which their unique attributes can shine. It also sounds kind of arbitrary to justify the decision by saying the outfit looks better, or to claim that the faction just prefers humans. Like, im not entirely against any instance of themed faction, but like, Im not sure why they just would prefer humans here.

For me I feel like it'd just make a whole lot more sense from an IC standpoint if the Spider Clan had rooted from hyper tech-maniacal humans, as in a lot of fantasy settings they are portrayed as the most "massive" and "prevalent" species due to their often early technological advancements, and the Spider Clan is what happens when instead of wanting to diversify into the cosmos the humans develop too much for their own good, and eventually turn into something cult-like (in this case, the worshipped idol is technology), it's also that much more unique, having a species-locked faction I mean, similarly to how in SS13 lore S.E.L.F. is mainly constitued of IPCs

ArtisticRoomba commented:
> > I want to repeat this as a comment because I dont know if it is visible as a pending review, but on the mention of all ninjas being humans:
> 
> > 
> 
> > I dont really see the point in making all of these species who are interesting in their own way and then not using them for antags, an entire class of role in which their unique attributes can shine. It also sounds kind of arbitrary to justify the decision by saying the outfit looks better, or to claim that the faction just prefers humans. Like, im not entirely against any instance of themed faction, but like, Im not sure why they just would prefer humans here.
> 
> 
> 
> For me I feel like it'd just make a whole lot more sense from an IC standpoint if the Spider Clan had rooted from hyper tech-maniacal humans, as in a lot of fantasy settings they are portrayed as the most "massive" and "prevalent" species due to their often early technological advancements, and the Spider Clan is what happens when instead of wanting to diversify into the cosmos the humans develop too much for their own good, and eventually turn into something cult-like (in this case, the worshipped idol is technology), it's also that much more unique, having a species-locked faction I mean, similarly to how in SS13 lore S.E.L.F. is mainly constitued of IPCs
> 
> 
> 
> Edit: to add onto this, Spider Clan's agents being standardized into bio-engineered humans also emphasizes the ninja's abilities and mechanics greatly, as humans by themselves are incredibly bland, there's no longer a "I'm used to this species" scenario (which I've personally come to develop as a player, I have a bad habit of constructing "safe" strategies around the usage of my species, instead of sometimes branching out with whatever tools I have at my disposal)

I disagree with both points.

I don't think it's a good excuse to disallow characters of other species strictly in terms of established IC lore, which isn't even present on upstream. Even if we accepted this, it makes more sense from an IC standpoint to accept other species strictly due to their benefits than average humans. Disregard this, I think that people want to be at least partially recognizable. All ninjas who just... look the same but behave differently is not fun, some people play their antag ghost roles under a certain gimmick established by them.

For the second point, I don't think this alone is enough to justify only accepting humans. I don't think players build their game plan around their species and get distracted by their species abilities. People are attracted to the ninjas abilities a thousand times more than they are attracted to two extra pocket slots or tail dragging. There's no need to force someone to play human in order to encourage the player to use their ninja abilities.

Dragonjspider commented:
Personally, I don't care either way if the role gets restricted to humans only or not, but I do have one issue regarding the ninja when it comes to the character you play, I have seen multiple times where the ninja played their character, but instead of being Nanotrasen aligned, they're Spider Clan, which makes sense, but it opens up a *very confusing* scenario, granted rare, where someone who died IC, in round, manages to get the role, and now... they're playing as their character, who died, as a ninja.

Now, this wouldn't be much of an issue, if it wasn't for the ghost role rules specifically stating that "You are absolutely not allowed to remember, say, the name, appearance, etc. of your previous character.
![Screenshot 2024-12-03 220215](https://github.com/user-attachments/assets/25e0d866-95aa-4d58-b37d-40d067693d01)

Which is an issue, because, in every way but name (Probably not DNA or Prints), you ARE the previous character if you died on that shift.

Related to that, it also means that players who you have a lot of interaction with in rounds would recognize you from your details, like your hair, the color or design of your tail, or maybe the color and design of your wings. Which, again, wouldn't be much of an issue, but this indirectly encourages people to regard you as *that very same character*, which in turn, makes it harder to disconnect the ninja from 'The guy I see on station every other shift'

So from my point of view, one of two things need to happen.
1. The role needs to randomize your character when spawned entirely
2. That rule specifically needs to be lifted for the ninja, letting the player be in character, while keeping the role, if they feel like it.

While yes, this isn't entirely related to the debate, I figured I would bring this up regardless, since it might change a bit about the situation.

Warpzoned commented:
> > > I want to repeat this as a comment because I dont know if it is visible as a pending review, but on the mention of all ninjas being humans:
> > 
> > 
> > > 
> > 
> > 
> > > I dont really see the point in making all of these species who are interesting in their own way and then not using them for antags, an entire class of role in which their unique attributes can shine. It also sounds kind of arbitrary to justify the decision by saying the outfit looks better, or to claim that the faction just prefers humans. Like, im not entirely against any instance of themed faction, but like, Im not sure why they just would prefer humans here.
> > 
> > 
> > For me I feel like it'd just make a whole lot more sense from an IC standpoint if the Spider Clan had rooted from hyper tech-maniacal humans, as in a lot of fantasy settings they are portrayed as the most "massive" and "prevalent" species due to their often early technological advancements, and the Spider Clan is what happens when instead of wanting to diversify into the cosmos the humans develop too much for their own good, and eventually turn into something cult-like (in this case, the worshipped idol is technology), it's also that much more unique, having a species-locked faction I mean, similarly to how in SS13 lore S.E.L.F. is mainly constitued of IPCs
> > Edit: to add onto this, Spider Clan's agents being standardized into bio-engineered humans also emphasizes the ninja's abilities and mechanics greatly, as humans by themselves are incredibly bland, there's no longer a "I'm used to this species" scenario (which I've personally come to develop as a player, I have a bad habit of constructing "safe" strategies around the usage of my species, instead of sometimes branching out with whatever tools I have at my disposal)
> 
> I disagree with both points.
> 
>...I think that people want to be at least partially recognizable. All ninjas who just... look the same but behave differently is not fun, some people play their antag ghost roles under a certain gimmick established by them.

I don't think behaving differently whilst looking similarly to be an issue, as the crew shouldn't be having thoughts like "oh! I recognize that character! It's (player)" to begin with, as a ninja you are to assume a (imho) completely separate persona to that of your usual characters, as it's a massively antagonistic opt-in ghost role, nuclear operatives (not loneops)are exempt from this even though they are massively antagonistic whilst using the player's characters since contrary to ninjas, they're rolled for roundstart, hence there not being any IC conflicts with them regarding lookalikes (eg. with ninja: them looking eerily similar to the cargo technician that just cryo'd). This is also an issue with loneops.

> For the second point, I don't think this alone is enough to justify only accepting humans. I don't think players build their game plan around their species and get distracted by their species abilities. People are attracted to the ninjas abilities a thousand times more than they are attracted to two extra pocket slots or tail dragging. There's no need to force someone to play human in order to encourage the player to use their ninja abilities.

I might have used an example too subjective and based on personal experience, for that I apologize for my lack of clarity 
- however I think implementing a moderately sudden shift to how a humanoid role plays out (kinda similar to how borgs play so differently to normal crew because of their boons and restrictions, even though in practice they mostly just do what the crew usually does) to be incredibly fun, and I don't want being able to "pick" your character to have any conflicts with this, as, going back to my first point, I want ninjas to assume a completely separate persona to that of the player's usual characters.

if my view of this isn't clear enough yet please do comment since wrapping my head around a matter this subjective is a toughie, lol.

Warpzoned commented:
> Personally, I don't care either way if the role gets restricted to humans only or not, but I do have one issue regarding the ninja when it comes to the character you play, I have seen multiple times where the ninja played their character, but instead of being Nanotrasen aligned, they're Spider Clan, which makes sense, but it opens up a _very confusing_ scenario, granted rare, where someone who died IC, in round, manages to get the role, and now... they're playing as their character, who died, as a ninja.
> 
> Now, this wouldn't be much of an issue, if it wasn't for the ghost role rules specifically stating that "You are absolutely not allowed to remember, say, the name, appearance, etc. of your previous character. ![Screenshot 2024-12-03 220215](https://private-user-images.githubusercontent.com/140023808/392221656-25e0d866-95aa-4d58-b37d-40d067693d01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMzMDYzMjYsIm5iZiI6MTczMzMwNjAyNiwicGF0aCI6Ii8xNDAwMjM4MDgvMzkyMjIxNjU2LTI1ZTBkODY2LTk1YWEtNGQ1OC1iMzdkLTQwZDA2NzY5M2QwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNFQwOTUzNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNGY4OWNlMTdhOTgwOWU5M2YyMzMzM2FlNjY2M2FiMThmODNiMWNhNGJjOTJlYjcxNDZmYzgyMTQ3NmRmMzcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.JgvAndNOZvHvFFUG0fVNb6YD-FHuqqgJ7XEGKRSXCFc)
> 
> Which is an issue, because, in every way but name (Probably not DNA or Prints), you ARE the previous character if you died on that shift.
> 
> Related to that, it also means that players who you have a lot of interaction with in rounds would recognize you from your details, like your hair, the color or design of your tail, or maybe the color and design of your wings. Which, again, wouldn't be much of an issue, but this indirectly encourages people to regard you as _that very same character_, which in turn, makes it harder to disconnect the ninja from 'The guy I see on station every other shift'
> 
> So from my point of view, one of two things need to happen.
> 
> 1. The role needs to randomize your character when spawned entirely
> 2. That rule specifically needs to be lifted for the ninja, letting the player be in character, while keeping the role, if they feel like it.
> 
> While yes, this isn't entirely related to the debate, I figured I would bring this up regardless, since it might change a bit about the situation.
> 
> (I made an issue about this since this doc reminded me of it [space-wizards/space-station-14#33713](https://github.com/space-wizards/space-station-14/issues/33713))

I... actually really like this way of addressing the issue! (regarding proposal 1.) But wouldn't that imply that there should be a shortened species selection to randomly pick from? (since a diona ninja kills off half of the things that makes it good)

Warpzoned committed:
better formatting, added comments, changed some objective stuff

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+
+- All stealing objectives can now only be performed with a specialized item that is dropped by the ninja upon death, unlike previously where they only used their gloves - This allows the crew to regain their losses.
+- Make an announcement - The communications computer objective now revolves around the Space Ninja sending a Spider Clan Announcement from it, the contents of said announcement being up to their discretion. Like before, it requires the use of turned on space ninja gloves, the one key difference being that now the EMAG interaction is instantaneous, overriding all access restrictions as well as unl ...
+- Hack X - 1-3 of the Miscellaneous Interactions machines is picked to be hacked by the Space Ninja. Note that you can still hack machines that aren't included in your objectives. (Replaces "Set Everyone to Wanted")
```
I like this idea.

While Ninjas aren't under the meta shield on WizDen, players still behave an in odd IC way and start to defend consoles and ninja objectives, even though it's much more interesting to pretend otherwise.

I think giving ninja enough objectives will be good, as security *possibly* can't guard them all. Security will have to take a more active approach to ninja countering. They can't simply spread out and guard everything, and if they try to, the ninja gets something unguarded in return.

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+
+- Steal X spesos - A strongbox for Cargo containing all their deposited spesos will be located in the vault, the Space Ninja is tasked with stealing a designated amount - An alternative to "Steal X technologies".
+- Upload the law board - An AI law board with a Spider Clan-oriented law set, the Space Ninja is tasked with uploading it to the AI upload console - An alternative to "Detonate your spider charge".
+- Hack Si-XXXX - The Space Ninja's kill-target equivalent, they are tasked with hacking a specific cyborg, turning it into a zealous Spider Clan tech-priest that obsesses over hoarding all manners of electronics and machines.
```
This is good. The SS13 implementation was transforming a Borg into some psychotic kill machine, but having a Borg that behaves similar to a ninja would be cool.

A suggestion to this point: Borgs should be able to hack the machines the same way a ninja would, albeit at a slower rate, and when completed, they count for the ninja. Might be something interesting to see.

Warpzoned replied:
> A suggestion to this point: Borgs should be able to hack the machines the same way a ninja would, albeit at a slower rate, and when completed, they count for the ninja. Might be something interesting to see.

I like this idea a lot, my only issue with it is that it''d probably be _way_ too good, I mean, an AA buddy that's not immediately valid? sounds like a dream! maybe the borg itself could get objectives that don't conflict with the ninja's by serving as a distraction? or maybe a "seppuku" module similar to the martyrs that the ninja can activate at will? open for more suggestions

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+
+All Space Ninjas will now have a Spider Clan Code they will have to live by, and this thus affects gameplay substantially by imposing certain restrictions.
+
+- The already existing one, being the inability to shoot ranged weapons, will be included in this Code.
```
Ninjas should get a roundstart Scroll of the Spider Clan Code, outlining an "honor code" that promotes treating enemies with respect and similar things (not RRing them for example). New players playing ninja might change their play style to fit this honor code given they are playing a ninja character, reducing murderboning. Obviously someone can ignore this, but I think it would immerse people in the role of a ninja more.

Warpzoned replied:
> Ninjas should get a roundstart Scroll of the Spider Clan Code, outlining an "honor code" that promotes treating enemies with respect and similar things (not RRing them for example). New players playing ninja might change their play style to fit this honor code given they are playing a ninja character, reducing murderboning. Obviously someone can ignore this, but I think it would immerse people in the role of a ninja more.

I was discussing the honor code mechanics with chief funk officer the other day, and it eventually devolved into a pretty funny idea that was as follows: basically, if the ninja kills too many personnel, eventually there'll be a "Son of X (killed victim)" that's tasked with finding and killing the ninja at any costs, a sort of terminator, and if said "Son of X" is killed there'll be a "Son of Son of X", so on and so forth.

Personally, I found this hilarious and would love it or something similar to be added, however, I think a more appropriate approach was for there to be a DeathSquad equivalent in the Spider Clan that if the ninja kills too many people one spawns in as a ghost raffle, and they have one and only objective: to kill the ninja, as they have tainted their code of honour, afterwards, they teleport back to base (acidify).

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+- Hacking the criminal records computer is now one of the many miscellaneous actions the Space Ninja can do in order to frustrate the crew - The error announcement will now only be sent by the criminal records computer on the Security channel, opposed to the previous station-wide Central Command Announcement.
+- The crew monitoring server is now prone to being hacked by the Space Ninja, doing so will disable everyone's sensors until deconstructed and reconstructed.
+- The station records computer is now prone to being hacked by the Space Ninja, doing so will delete a small % of random personnel from it accompanied by a station-wide announcement.
+- The camera router is now prone to being hacked by the Space Ninja, doing so will kill a large % of its connected cameras and render them useless.
```
I'd like the server to become unrepairable, and require an engineer or so to do maintenance on the camera router. It can be something where there isn't no true physical destruction, it's just disrupted and required repair.

This would make countering the AI _extremely viable_ and the ninja could plan ahead: "I'll break into telecoms and short out the cameras to security, so I can get a high-cap without being spotted"

To reduce metagaming ("ninja on station, my cameras in sec aren't working!!") we could have this same effect trigger for a random telecoms server during a solar flare. It would expose players to the situation more often and make it less unique to the ninja. EMPs should also disrupt cameras, and if they hit servers, it should disrupt them and cut cams too. I believe there's an issue on the repo right now that asks the question "should cameras have to be routed through a router" which I really like, both for regular gameplay, and for this idea.

Warpzoned replied:
> I'd like the server to become unrepairable, and require an engineer or so to do maintenance on the camera router. It can be something where there isn't no true physical destruction, it's just disrupted and required repair.

Maybe, just maybe, what if instead of the router itself being affected only the cameras require repairing? It'd probably require for a certain % of them to suddenly get their wires cut, though. I just can't really see any way of the engineers having a way to repair routers without it just being a "use in-case of ninja" thing.

> This would make countering the AI _extremely viable_ and the ninja could plan ahead: "I'll break into telecoms and short out the cameras to security, so I can get a high-cap without being spotted"
> 
> To reduce metagaming ("ninja on station, my cameras in sec aren't working!!") we could have this same effect trigger for a random telecoms server during a solar flare. It would expose players to the situation more often and make it less unique to the ninja. EMPs should also disrupt cameras, and if they hit servers, it should disrupt them and cut cams too. I believe there's an issue on the repo right now that asks the question "should cameras have to be routed through a router" which I really like, both for regular gameplay, and for this idea.

agreed! it'd also enrich engineering gameplay which they severely need, also, if possible, could you link said question so I can take a look at the discussions that take place?

deltanedas replied:
cameras dont even need to be set up for AI to see through them, let alone need a working router

Warpzoned replied:
> cameras dont even need to be set up for AI to see through them, let alone need a working router

yeah that's pretty weird, though Isn't this mostly due to AI atm being extremely barebones? Plus, this is also troublesome for people like the Warden, who are also constantly checking cameras.

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+These nerfs are intended to make previous go-to combat tactics less prevalent, hence incentivizing a more varied playstyle.
+
+- [Fix ninja stunlocking](https://github.com/space-wizards/space-station-14/pull/33244)
+- All Space Ninjas are now bio-engineered humans, making it so they don't get specie-specific attributes such as tail-dragging.
```
Having to restrict an antags species to human is lame and circumvents a whole part of the game: different species. It's part of this games character.

See: SS13 where nukies were only humans because the AI was speciesist. We can do better in SS14.

Warpzoned replied:
> Having to restrict an antags species to human is lame and circumvents a whole part of the game: different species. It's part of this games character.
> 
> See: SS13 where nukies were only humans because the AI was speciesist. We can do better in SS14.

I had a relatively discussion about this on discord recently (after the comments one), and really I think I'll be implementing Dragonjspider's "The role needs to randomize your character when spawned entirely" proposal, as it'd be the best of both worlds imho, having it be your character could prove not nearly as fun as if you had done it with a different species, but having always be fixed to a certain species could prove to be against the game's character, and so it'll pick a species from a certain species pool (eg. humans, dwarves, reptillians, arachnids - as they are moderately "standard")

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+
+Mostly new additions that weren't either complete buffs nor complete nerfs.
+
+- The katana dash is considerably quieter, at the cost of a longer recharge duration.
```
Im ok with this, as it gives the Ninja more of a chance to slip up and make a mistake. Right now a really good ninja is untouchable by security because of their amazing ability to pick fights. However, despite this change, I still think a very robust player can still escape 99% of situations.

Warpzoned replied:
> Im ok with this, as it gives the Ninja more of a chance to slip up and make a mistake. Right now a really good ninja is untouchable by security because of their amazing ability to pick fights. However, despite this change, I still think a very robust player can still escape 99% of situations.

this is more as a way to give the ninja a proper quieter way to move around, this time along the dashes are made to be more silent, as the "loud" action is the EMAG (as it's also the more overpowered one), so the ninja can pick between either having AA access but going loud, or dashing from window to window stealthily

ArtisticRoomba commented on src/en/space-station-14/round-flow/proposals/ninja.md:
```diff
@@ -0,0 +1,94 @@
+Mostly new additions that weren't either complete buffs nor complete nerfs.
+
+- The katana dash is considerably quieter, at the cost of a longer recharge duration.
+- Grants the ability to wield the energy katana for a hard-hitting swing, at the cost of a slower slash rate. *When wielding the energy katana, the Space Ninja's reflect probability also increases.
```
There's a lot of different things you could do with this:

Have the sword reflect an insane amount if the ninja keeps pointing at the attacker. Because, you know, the ninja can parry. The downside to this is that the ninja is exposed from all other sides: shooting the ninja is no longer just a gamble on whether you score a hit or not.

I think the reflect system could also be refactored to determine the reflect chance of lasers versus bullets, that way you could give the sword a much higher chance to reflect bullets as they travel slower than lasers that travel at the speed of light. This could also make balancing items that can reflect easier.

Warpzoned replied:
> There's a lot of different things you could do with this:
> 
> Have the sword reflect an insane amount if the ninja keeps pointing at the attacker. Because, you know, the ninja can parry. The downside to this is that the ninja is exposed from all other sides: shooting the ninja is no longer just a gamble on whether you score a hit or not.

Interesting way to go about it, but it'd probably require some heavy changes to the reflect system in general to add things related to orientation. I'll think about it, however.

> I think the reflect system could also be refactored to determine the reflect chance of lasers versus bullets, that way you could give the sword a much higher chance to reflect bullets as they travel slower than lasers that travel at the speed of light. This could also make balancing items that can reflect easier.

Speaking of heavy changes, I agree with this wholeheartedly, lasers should be prone to reflection via stuff like appropriate vests or mirrors, and blades should also have an absorb chance when it comes to bullets, the latter which I think would work wonder for the ninja (basically cutting bullets in half)

TeenSarlacc replied:
> Have the sword reflect an insane amount if the ninja keeps pointing at the attacker.

100% bullet reflect while moving max speed so you can pretend you're playing mgrr :trollface:

Warpzoned replied:
> > Have the sword reflect an insane amount if the ninja keeps pointing at the attacker.
> 
> 100% bullet reflect while moving max speed so you can pretend you're playing mgrr :trollface:

.1% probability for the ninja to have one of jetstream sam's voicelines after a successful reflection ![image](https://github.com/user-attachments/assets/c43100f3-623f-42cc-b26b-08080cd7c036)

ArtisticRoomba reviewed (commented):
Left comments on your recent changes as per your request.

Warpzoned committed:
part 1/2 of the final update

Warpzoned committed:
final update 2/2

hurray !

Pumkin69 commented:
I don't think ninja should have a stun at all to be honest i don't think it fits the gameplay that we want ninja to facilitate and it literally just leads to murderboning because stuns are so op.

Princess-Cheeseballs commented:
Docs meeting went over this design doc and has decided to close this PR. It focuses much more on balance changes and tweaks over describing the aesthetics and goals of the space ninja as an antagonist. As such it's very much tied to current implementation and not future proofed leading to the decision to close in favor of #547
